homeassistant/components/growatt_server/sensor.py

```diff
@@ -558,17 +558,22 @@
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Growatt sensor."""
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-    plant_id = config[CONF_PLANT_ID]
-    name = config[CONF_NAME]
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up growatt server from yaml."""
+    if not hass.config_entries.async_entries(DOMAIN):
```

Comment:
Please log a warning that the configuration.yaml is deprecated. See eg:
https://github.com/home-assistant/core/blob/ba31d7d1b493aaec4cce27f0cb243ab52e094ce6/homeassistant/components/onewire/sensor.py#L239-L242